src/styles/stars.scss

```diff
@@ -0,0 +1,24 @@
+@use './variables' as *;
+
+.stars {
+  display: flex;
+
+  &__star {
+    background-image: url(../images/star.svg);
+    background-position: center;
+    background-repeat: no-repeat;
+    height: $star-size;
+    width: $star-size;
+    margin-right: $star-margin;
+
+    &:last-child {
+      margin-right: 0;
+    }
+  }
+
+  @for $i from 1 through 5 {
+    &--#{$i} :nth-child(-n + #{$i}) {
```

Comment:
The selector `&--#{$i} :nth-child(-n + #{$i})` might not work as expected. The space between `&--#{$i}` and `:nth-child` will result in a descendant combinator, which is likely not the intended behavior. If you want to apply styles to the nth-child of the element with the modifier class, remove the space: `&--#{$i}:nth-child(-n + #{$i})`.